### ISSUE TITLE
HttpUtility.JavaScriptStringEncode single quote behavior does not match documentation

### ISSUE THREAD

**Issue by ogjkoch** [dotnet/runtime#100248]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

The [documentation](https://learn.microsoft.com/en-us/dotnet/api/system.web.httputility.javascriptstringencode?view=net-8.0) for HttpUtility.JavaScriptStringEncode states that the single quote character (') will be escaped as \\'.

However, it gets escaped as \u0027.

Double quote is escaped as the documentation describes.

### Expected Behavior

Expected that HttpUtility.JavaScriptStringEncode functions as documented.

### Steps To Reproduce

Pass a string containing a single quote character (') to HttpUtility.JavaScriptStringEncode.

[Here](https://dotnetfiddle.net/8F4IDx) is a .NET fiddle displaying this behavior.

### .NET Version

8.0

**Comment by MihaZupan** [comment]
I think the behavior here is correct, `\'` isn't a valid escape sequence.
The docs appear to be wrong here.

**Comment by amcasey** [comment]
> I think the behavior here is correct, `\'` isn't a valid escape sequence. The docs appear to be wrong here.

Do you have more details on this behavior?  When I type `"\'"` in the browser console, it outputs `"'"`.  Is that a non-compliant implementation?  Naively, I would have guessed it was allowed simply because JS allows `'`-delimited strings.

Or maybe I've misunderstood entirely and we're talking about C# string literals?

**Comment by MihaZupan** [comment]
Ah, looks like I was looking at JSON, which defines the same set except for `'`.
Assuming the Unicode escape sequence `\u0027` would be treated the same way as `\'`, do you think updating the docs here would be sufficient? 
It seems plausible that people are using this API for JSON string escaping, and `\'` would break those.

**Comment by amcasey** [comment]
> It seems plausible that people are using this API for JSON string escaping, and \' would break those.

While that does seem plausible, the code already escapes `'`.  Changing that seems likely to break people using the type to escape JS strings.  It may, however, be worth putting a remark in the docs about having to consider `\'` if you're escaping for JSON.

**Comment by MihaZupan** [comment]
Right, I'm suggesting we don't change the behavior here and update the docs instead since over-escaping is fine in JSON (and should be in JS strings as well?).

**Comment by amcasey** [comment]
My turn to get mixed up. 😆  I forgot what the current state was.

Yes, I think I'd agree that a doc update makes more sense.  I kind of wish we had escaped single quote "properly" in the first place, but that ship has sailed.

**Comment by MihaZupan** [comment]
Done in https://github.com/dotnet/dotnet-api-docs/pull/9901
